src/components/frame/v5/pages/BalancePage/partials/BalanceTable/hooks.tsx

```diff
@@ -171,3 +176,137 @@ export const useGetTableMenuProps = () =>
       ],
     };
   }, []);
+
+export const useBalanceTable = () => {
+  const { colony } = useColonyContext();
+  const selectedTeam = useGetSelectedTeamFilter();
+  const { balances, nativeToken } = colony || {};
+  const [searchValue, setSearchValue] = useState('');
+  const [value, setValue] = useState<Partial<BalanceTableFilters>>({
+    type: {},
+    attribute: {
+      native: false,
+      reputation: false,
+    },
+  });
+
+  const tokensData = useMemo(
+    () =>
+      colony?.tokens?.items.map((item) => {
+        const currentTokenBalance =
+          getBalanceForTokenAndDomain(
+            balances,
+            item?.token?.tokenAddress || '',
+            selectedTeam ? Number(selectedTeam.nativeId) : undefined,
+          ) || 0;
+        const decimals = getTokenDecimalsWithFallback(item?.token.decimals);
+        const convertedValue = convertToDecimal(
+          currentTokenBalance,
+          decimals || 0,
+        );
+
+        const formattedValue = getFormattedNumeralValue(
+          convertedValue,
+          currentTokenBalance,
+        );
+
+        return {
+          ...item,
+          balance: typeof formattedValue === 'string' ? formattedValue : '',
+        };
+      }),
+    [colony, balances, selectedTeam],
+  );
+
+  const sortedTokens = useMemo(
+    () =>
+      tokensData?.sort((a, b) => {
+        if (!a.balance || !b.balance) return 0;
+        return parseInt(b.balance, 10) - parseInt(a.balance, 10);
+      }),
+    [tokensData],
+  );
+
+  const tokenTypeFilters = sortedTokens
```

Comment:
we shouldn't return JSX from hooks, the comment I dropped about composition would probably solve this